ImageProcessing/BrighterFatterCorrection.ipynb

```diff
@@ -0,0 +1,767 @@
+    "\n",
+    "# choose which stack measurements to use for centroids and shape\n",
+    "# +TODO use 'ext_shapeHSM_HsmSourceMoments_xx','ext_shapeHSM_HsmSourceMoments_yy'\n",
+    "cen_param1,cen_param2='base_SdssCentroid_x','base_SdssCentroid_y'\n",
+    "bf_param1,bf_param2='base_SdssShape_xx','base_SdssShape_yy'\n",
+    "flux_param='base_SdssShape_flux'\n",
+    "\n",
+    "# get the centroids (used for matching) from the fiducial frame \n",
+    "x0s,y0s=cat_arr[fidframe][cen_param1],cat_arr[fidframe][cen_param2]\n",
+    "nspots=len(x0s)\n",
+    "\n",
+    "# make an array to hold that number of objects and their centroid/shape/flux measurements\n",
+    "# the 8 rows collect x/y centroid, x/y shape, x/y corrected shape, flux, and corrected flux\n",
+    "bf_dat=np.empty((ncats,nspots,8))\n",
+    "bf_dat[:]=np.nan   # so that un-matched objects aren't plotted/used by default\n",
+    "\n",
+    "\n",
+    "# loop over catalogs\n",
+    "for i in range(ncats):\n",
+    "    # get the centroids of objects in the bf-corrected and uncorrected images\n",
+    "    x1,y1=cat_arr[i][cen_param1],cat_arr[i][cen_param2]\n",
+    "    x1_bf,y1_bf=bf_cat_arr[i][cen_param1],bf_cat_arr[i][cen_param2]\n",
+    "    # loop over fiducial frame centroids to find matches\n",
+    "    for j in range(nspots):   \n",
+    "        x0,y0=x0s[j],y0s[j] # fiducial centroid to match\n",
+    "        # find objects in both catalogs which are within maxdist\n",
+    "        bf_gd=np.where(np.sqrt((x1_bf-x0)**2+(y1_bf-y0)**2)<maxdist)[0]\n",
+    "        gd=np.where(np.sqrt((x1-x0)**2+(y1-y0)**2)<maxdist)[0]\n",
+    "        if (len(bf_gd)==1 & len(gd)==1):  # only take single matches\n",
+    "            xx,yy=cat_arr[i][bf_param1][gd],cat_arr[i][bf_param2][gd] # centroids\n",
+    "            xx_bf,yy_bf=bf_cat_arr[i][bf_param1][bf_gd],bf_cat_arr[i][bf_param2][bf_gd]  # sizes\n",
+    "            flux,flux_bf=cat_arr[i][flux_param][gd],bf_cat_arr[i][flux_param][bf_gd] # fluxes\n",
+    "            bf_dat[i,j,:]=x0,y0,xx,yy,xx_bf,yy_bf,flux,flux_bf  # keep those above measurements"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Plot the brighter-fatter effect on those shape measurements and the corrected version\n",
+    "Alongside stamps of each object, below we show the trend of X and Y sizes before and after brighter-fatter correction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sz=11 # stamp size\n",
+    "\n",
+    "# loop over some objects and save a summary figure\n",
+    "# [0,6,12,18,23,35,44,46,52,56,69,71,114] are good indices to look with default values \n",
+    "# or e.g. np.random.choice(np.arange(nspots),size=10)\n",
+    "indexfoo=[0,6,12,18,23,35,44,46,52,56,69,71,114]\n",
+    "\n",
+    "for nfoo in indexfoo:\n",
+    "    plt.figure(figsize=(14,4)),plt.subplots_adjust(wspace=.3)\n",
```

Comment:
Noted that this could be due to violation of kernel correction assumptions and/or signal to noise increases, however the latter is a universal issue with shape measurement and is beyond the scope of the notebook.